api/server/router/image/image_routes.go

```diff
@@ -153,7 +153,25 @@ func (ir *imageRouter) postImagesPush(ctx context.Context, w http.ResponseWriter
 
 	img := vars["name"]
 	tag := r.Form.Get("tag")
-	if err := ir.backend.PushImage(ctx, img, tag, metaHeaders, authConfig, output); err != nil {
+
+	var ref reference.Named
+
+	// Tag is empty only in case ImagePushOptions.All is true.
+	if tag != "" {
```

Comment:
I just recalled we still have an open ticket for this to change this at the API level as well;

- https://github.com/moby/moby/pull/40302 changed the client to require an explicit opt-in for "all tags"
- https://github.com/docker/cli/pull/2220 changed the CLI to use it
- https://github.com/moby/moby/issues/42053has a proposal to change the API to have an explicit `all-tags` options

Maybe we should start working on making that API change 🤔 (we could mention the linked ticket here as a TODO)